exporter/logzioexporter/exporter.go

```diff
@@ -15,9 +15,96 @@
 package logzioexporter
 
 import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/logzio/jaeger-logzio/store"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"go.opentelemetry.io/collector/translator/trace/jaeger"
+)
+
+const (
+	loggerName = "logzio-exporter"
 )
 
+// logzioExporter implements an OpenTelemetry trace exporter that exports all spans to Logz.io
+type logzioExporter struct {
+	accountToken string
+	writer       *store.LogzioSpanWriter
+	logger       hclog.Logger
+}
+
+var WriteSpanFunc func(span *model.Span) error
+var InternalTracesToJaegerTraces = jaeger.InternalTracesToJaegerProto
+
+func newLogzioExporter(config *Config, params component.ExporterCreateParams) (*logzioExporter, error) {
+	logger := Hclog2ZapLogger{
+		Zap:  params.Logger,
+		name: loggerName,
+	}
+
+	if config == nil {
+		return nil, errors.New("exporter config can't be null")
+	}
+	writerConfig := store.LogzioConfig{
+		Region:            config.Region,
+		AccountToken:      config.Token,
+		CustomListenerURL: config.CustomEndpoint,
+	}
+
+	spanWriter, err := store.NewLogzioSpanWriter(writerConfig, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &logzioExporter{
+		writer:       spanWriter,
+		accountToken: config.Token,
+		logger:       logger,
+	}, nil
+}
+
 func newLogzioTraceExporter(config *Config, params component.ExporterCreateParams) (component.TraceExporter, error) {
-	return nil, nil
+	exporter, err := newLogzioExporter(config, params)
+	if err != nil {
+		return nil, err
+	}
+	WriteSpanFunc = exporter.writer.WriteSpan
```

Comment:
Would be better to make this a field on the exporter struct instead of a global variable.  Then your tests can still override it.

Reply:
You will need to return `*logzioExporter` from this func and then do the `exporterhelper` stuff in the factory to be able to access things properly in tests.  See the `otlpexporter` in core for an example.